### PR TITLE
CORE-1299: Extend the Core Demo App with Wallet Connect

### PR DESCRIPTION
**Pull request by bgoring**

### CODE REVIEW

File: WalletKitCore/src/walletkit/handlers/eth/WKWalletConnectorETH.c [hunk 1]
```diff
@@ -147,13 +150,15 @@ wkWalletConnectorSignDataETH (
     // Update the signature `V` field with the EIP-155 encoding as per https://eips.ethereum.org/EIPS/eip-155
     //    "the v of the signature MUST be set to {0,1} + CHAIN_ID * 2 + 35 where {0,1} is the parity of y"
     // Because the VRS_EIP signing produces `v` of {27|28}, we transfer the EIP-155 encoding to:
+#ifndef WALLET_CONNECT_1_0_TESTING
     signature.sig.vrs.v += (8 + 2 * chainId);
+#endif
```
Comment (EBGToo) [inline WalletKitCore/src/walletkit/handlers/eth/WKWalletConnectorETH.c:155]: I'd prefer if the mainline code not depending on 'TESTING'.  Perhaps:

```
#define NEED_EIP_155_SIGNATURE_ENCODING
...
#if defined (NEED_EIP_155_SIGNATURE_ENCODING)
    signature.sig.vsr.v += ...
#endif
```

File: WalletKitCore/src/walletkit/handlers/eth/WKWalletConnectorETH.c [hunk 2]
```diff
@@ -522,7 +527,9 @@ wkWalletConnectorSignTypedDataETH (
     // Update the signature `V` field with the EIP-155 encoding as per https://eips.ethereum.org/EIPS/eip-155
     //    "the v of the signature MUST be set to {0,1} + CHAIN_ID * 2 + 35 where {0,1} is the parity of y"
     // Because the VRS_EIP signing produces `v` of {27|28}, we transfer the EIP-155 encoding to:
+#ifndef WALLET_CONNECT_1_0_TESTING
     signResult.signature.sig.vrs.v += (8 + 2 * chainId);
+#endif
```
Comment (EBGToo) [inline WalletKitCore/src/walletkit/handlers/eth/WKWalletConnectorETH.c:532]: Ditto

File: WalletKitJava/WalletKitDemo-Android/build.gradle [hunk 3]
```diff
@@ -59,6 +59,8 @@ dependencies {
     implementation deps.jacksonDatabind
     implementation deps.jacksonDatabindGuava
 
+    implementation deps.bouncyCastle
+
```
Comment (EBGToo) [inline WalletKitJava/WalletKitDemo-Android/build.gradle:63]: Perhaps preceded w/ a comment of `For WalletConnect`?

File: WalletKitJava/WalletKitDemo-Android/src/main/AndroidManifest.xml [hunk 4]
```diff
@@ -14,6 +14,7 @@
         android:roundIcon="@mipmap/ic_launcher_round"
         android:supportsRtl="true"
         android:theme="@style/AppTheme"
+        android:usesCleartextTraffic="true"
```
Comment (EBGToo) [inline WalletKitJava/WalletKitDemo-Android/src/main/AndroidManifest.xml:17]: '// For WalletConnet'

File: WalletKitCore/src/walletkit/handlers/eth/WKWalletConnectorETH.c (hunk 2 above)
Comment (EBGToo) [reply WalletKitCore/src/walletkit/handlers/eth/WKWalletConnectorETH.c:532]: Also, somewhere is `vrs.v -= ...` where it is decoded.  Something is needed there - if only a comment of 'handle both w/ and w/o EIP-155 encoding'

File: WalletKitCore/src/walletkit/handlers/eth/WKWalletConnectorETH.c (hunk 1 above)
Comment (bgoring) [reply WalletKitCore/src/walletkit/handlers/eth/WKWalletConnectorETH.c:155]: Yes, it's better this way. Incorporated this.

File: WalletKitJava/WalletKitDemo-Android/build.gradle (hunk 3 above)
Comment (bgoring) [reply WalletKitJava/WalletKitDemo-Android/build.gradle:63]: Done.

File: WalletKitJava/WalletKitDemo-Android/src/main/AndroidManifest.xml (hunk 4 above)
Comment (bgoring) [reply WalletKitJava/WalletKitDemo-Android/src/main/AndroidManifest.xml:17]: Actually may not even be needed unless testing against a local non https bridge. I'll check it and if not, just leave it commented with a suitable note.

File: WalletKitCore/src/walletkit/handlers/eth/WKWalletConnectorETH.c (hunk 2 above)
Comment (bgoring) [reply WalletKitCore/src/walletkit/handlers/eth/WKWalletConnectorETH.c:532]: In recovery:
`    // Undo the EIP-155 encoding if it exists.  See above `wkWalletConnectorSignDataETH()`.  Note:
    // we decode the `sigChainId` and compare it to `outChainId` to confirm a valid signature.
#if defined (NEED_EIP_155_SIGNATURE_ENCODING)
    if (vrs.v > 28) {
        BREthereumChainId sigChainId = (vrs.v - 35) / 2;

        validChainId = (sigChainId == ourChainId);
        if (validChainId) vrs.v -= (8 + 2 * sigChainId);
    }
#endif`

**Pull request merged**